src/components/Provider/index.tsx

```diff
@@ -22,7 +23,11 @@ const Base = styled(Grommet)`
 `;
 
 const Provider = ({ theme, ...props }: ThemedProvider) => {
-	return <Base theme={merge(defaultTheme, theme)} {...props} />;
+	return (
+		<BreakpointProvider breakpoints={defaultTheme.breakpoints}>
```

Comment:
We should pass the breakpoints from the merged theme here, not the default one. You would extract `merge(defaultTheme, theme)` in a variable, and use that to pass the breakpoints.